src/components/pages/History/PoolShares.tsx

```diff
@@ -56,20 +79,33 @@ export default function PoolShares(): ReactElement {
   const { ocean, accountId, config } = useOcean()
   const [assets, setAssets] = useState<Asset[]>()
   const [isLoading, setIsLoading] = useState(false)
+  const [pools, setPools] = useState([])
+  const { data } = useQuery<PoolShare>(poolSharesQuery, {
+    variables: {
+      user: accountId?.toLowerCase()
+    },
+    pollInterval: 20000
+  })
+
+  useEffect(() => {
+    if (!data) return
+    setPools(data.poolShares)
+  }, [data])
 
   useEffect(() => {
     async function getAssets() {
       if (!ocean || !accountId || !config?.metadataCacheUri) return
       setIsLoading(true)
 
       try {
-        const pools = await ocean.pool.getPoolSharesByAddress(accountId)
         const metadataCache = new MetadataCache(config.metadataCacheUri, Logger)
         const result: Asset[] = []
-
         for (const pool of pools) {
-          const ddo = await metadataCache.retrieveDDO(pool.did)
-          ddo && result.push({ ddo, shares: pool.shares })
+          const did = web3.utils
```

Comment:
minor performance hit here because you fetch all the ddos before and not on row but considering there are few items and we will change the way we fetch the name it's ok

Reply:
not sure this is that minor. When comparing the 2 tables on history page here https://market-git-feature-get-pool-shares-using-graphql.oceanprotocol.now.sh/history the pool shares table is significantly slower for me, although I have more items in the pool transactions than in the pool shares. Pool transactions come in in under 1 sec., pool shares take 10 sec. for me

Reply:
they happen at the same time for me, but I have just a few items, 2 pool shares and 3 pool transaction

Reply:
Then i underestimated the impact so it needs to be as it is on the tx table

Reply:
as it is now the table doesn't get just the data set name from the ddo but the total pool liquidity too, so if I implement it as in transactions component there would be 2 requests, one for the name column and one for the total pool liquidity. But if the total pool liquidity will be replaced with something else that we can take from pool shares than we can get the asset title only